### PR TITLE
WIP: Improve the performance of row extraction and table building

### PR DESCRIPTION
**Pull request by pm5k**
# TL;DR:

This is a PR to fix the slow performance of parsing rows and columns for large DAT tables, as well as some minor refactoring suggestions for the future.

# Changes:

- Slightly changed the way reading of columns is done. Instead of doing large numbers of assignment to awaitables in a loop, I used a Promise resolution to do it all at once ahead of time and then loop over the result. This showed a pow3~ improvement to loading a large dataset the first time around before it is cached (is it really cached..?)

# Caveats:

I found that there are other things that are wrong here, hence this is a WIP and not a mergeable PR unless @Journeytojah deems this is enough. If the PR is closed, move this to an issue:

- There is a double render happening when the table is constructed. You first see headers and "undefined data" in the cells, and then it populates. This implies the actual order of ops is a bit wonky and the entire table should probably be rewritten later.
- Subsequent loads of tables can inconsistently bring the slow loading back despite clear signs the initial load is fast. If I load "words.dat" initially it'll take 5-10ms to parse it, then loading a larger dataset (like the world areas) will take absolute ages, and then a subsequent load of "words.dat" will take the pre-fix time of 8s or so. This is not always occurring and requires digging into.

### CODE REVIEW

**Comment by pm5k** [comment]
Closing this as @Journeytojah pulled the perf code into his perf branch
